Create Clickhouse config

adityachopra29 opened the pull request:
## Which problem is this PR solving?
- Solves #6154 

## Description of the changes
- Create `plugin/storage/clickhouse/`
- This is a draft pr, not ready to be merged.

## How was this change tested?

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [ ] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `npm run lint` and `npm run test`

adityachopra29 commented:
This is a draft pr.
@yurishkuro  I am having trouble writing the schema config, as we dont have a schema ready right now.
Also wanted to confirm whether I am on the right track.

adityachopra29 commented:
Just realised a structure change is required. Will make it and update the pr shortly.

yurishkuro commented:
What is the motivation for the specific config parameters you are introducing? It feels a bit like putting the horse in front of the carriage - we don't have an implementation of the storage yet, so how do we know which config options will even be required?

zzzk1 commented on plugin/storage/clickhouse/config.go:
```diff
@@ -0,0 +1,102 @@
+}
+
+// BasicAuthenticator holds the username and password for a password authenticator for a Clickhouse cluster.
+type BasicAuthenticator struct {
+	Username string `mapstructure:"username"`
+	Password string `mapstructure:"password"`
+}
```
Only the following are required for a mini connection: 1. Host 2. Port 3. Database 4. Username 5. Password.

adityachopra29 replied:
Thanks!

zzzk1 reviewed (commented):
you can try: https://clickhouse.com/docs/en/integrations/go#connecting-1

adityachopra29 commented:
> What is the motivation for the specific config parameters you are introducing? It feels a bit like putting the horse in front of the carriage - we don't have an implementation of the storage yet, so how do we know which config options will even be required?

@yurishkuro I see. I looked through the configurations for the other db storages, and saw what types of configurations jaeger was using to model config for clickhouse. But adding config as we develop is more logical.
I have moved the config to `/pkg/storage/clickhouse` and also removed the extra config. Pls have a look
